### ISSUE TITLE
Java Program: run: line 1: 3 killed

### ISSUE THREAD

**Issue by NickCanzSGS** [thebaselab/codeapp#836]
Hello everyone!
This is my first time here in Git Hub so forgive me if this is not the appropriate section where you can share a problem (also sorry for my bad english). I'm a student and we are working on a software that can calculate the "Manhattan Tourist Problem" (if you don't know what this is you can easily search it on the web). 

This is the code:
import java.math.*;

public class ManhattanMemoization {

    public static final BigInteger BIG_UNKNOWN = new BigInteger("0");

    public static void main(String args[]) {
        int i = 20;

        System.out.println(manhattanM2(i, i));
    }

    public static BigInteger manhattanM2(int i, int j) {

        BigInteger[][] mem = new BigInteger[i+1][j+1];

        for (int h = 0; h <= i; h++) {

            for (int k = 0; k <= j; k++) {

                mem[h][k] = BIG_UNKNOWN;

            }

        }

        return manhattanMem2(i, j, mem);

    }

    public static BigInteger manhattanMem2(int i, int j, BigInteger[][] mem) {

        if (mem[i][j] == BIG_UNKNOWN) {

            if (i == 0 || j == 0) {

                mem[i][j] = new BigInteger("1");

            } else {

                return manhattanMem2(i-1, j, mem).add(manhattanMem2(i, j-1, mem));

            }

        }
        
        return mem[i][j];

    }

}

Basically the variable "i" stands for the amount of rows and columns in this hypothetical manhattan city. If I set the "i" value to 11 or less the software works fine but if I insert an higher value like 15 or even higher like 20 or 50 in the "output" section I have this error: run: line 1: 3 killed java ManhattanMemoization.
I think is a problem related to the amount of memory available (I'm on iPad Pro 2018) but the strange thing is that the day before everything works fine without this error.
Anyone know why I have this error???
Thanks anyway

**Comment by NickCanzSGS** [comment]
I solved the problem...
The problem was the return inside the else in the "manhattanMem2" method. I fixed using:
mem[i][j] = manhattanMem2(i-1, j, mem).add(manhattanMem2(i, j-1, mem));
